bwpatcher/core_lks32.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+#! -*- coding: utf-8 -*-
+#
+# BW Patcher
+# Copyright (C) 2024-2025 ScooterTeam
+#
+# This work is licensed under the Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License.
+# To view a copy of this license, visit http://creativecommons.org/licenses/by-nc-sa/4.0/
+# or send a letter to Creative Commons, PO Box 1866, Mountain View, CA 94042, USA.
+#
+# You are free to:
+# - Share — copy and redistribute the material in any medium or format
+# - Adapt — remix, transform, and build upon the material
+#
+# Under the following terms:
+# - Attribution — You must give appropriate credit, provide a link to the license, and indicate if changes were made.
+# - NonCommercial — You may not use the material for commercial purposes.
+# - ShareAlike — If you remix, transform, or build upon the material, you must distribute your contributions under the same license as the original.
+#
+
+import math
+
+from bwpatcher.core import CorePatcher
+from bwpatcher.utils import find_pattern, SignatureException
+
+
+class LKS32MC07Patcher(CorePatcher):
+    def __init__(self, data):
+        super().__init__(data)
+
+    def _speed_limits_fix(self):
+        ret = []
+        sig = [0x59, 0x68, None, 0x4A, None, 0x3A, 0x91, 0x42]
+        sig_dst = [0xF5, 0x31, 0x41, 0x81, 0x70, 0xBD]
+
+        ofs = find_pattern(self.data, sig) + len(sig)
+        try:
+            # jump to the end of the function
+            ofs_dst = find_pattern(self.data, sig_dst, start=ofs) + 4
+            pre = self.data[ofs:ofs+2]
+            post = self.assembly(f"b {ofs_dst-ofs}")
+            self.data[ofs:ofs+2] = post
+            ret.append(("speed_limits_fix", hex(ofs), pre.hex(), post.hex()))
+
+            # NOP unnecessary code, making space for 4-bytes speed limit values
+            ofs += 2
+            nop_size = ofs_dst - ofs
+            assert nop_size % 2 == 0, "Odd size of the space needed."
+
+            pre = self.data[ofs:ofs+nop_size]
+            post = self.assembly(f"nop") * (nop_size//2)
+            assert len(pre) == len(post), "Wrong length of post bytes"
+
+            self.data[ofs:ofs+nop_size] = post
+            ret.append(("speed_limits_fix_nop", hex(ofs), pre.hex(), post.hex()))
+        except SignatureException:
+            # verify if the patch has already been used
+            # otherwise the function will raise a SignatureException
+            sig_dst = [0x00, 0xBF, 0x00, 0xBF, 0x70, 0xBD]
+            ofs_dst = find_pattern(self.data, sig_dst, start=ofs)
+
+        return ret
+
+    @staticmethod
+    def _safe_ldr(ldr_offset: int, dst_ofs: int) -> tuple[int, int]:

```

Comment:
I do something similar in 5 pro branch, let's evaluate later which approach is better